src/routes/safe/components/Apps/utils.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import axios from 'axios'
+
+import appsIconSvg from '~/routes/safe/components/Transactions/TxsTable/TxType/assets/appsIcon.svg'
+
+export const GNOSIS_APPS_URL = 'https://gnosis-apps.netlify.com'
+
+export const getAppInfoFromOrigin = (origin: string) => {
+  try {
+    return JSON.parse(origin)
+  } catch (error) {
+    console.error(`Impossible to parse TX origin: ${origin}`)
+    return null
+  }
+}
+
+export const getAppInfoFromUrl = async (appUrl: string) => {
+  try {
+    const appInfo = await axios.get(`${appUrl}/manifest.json`)
+    const res = { url: appUrl, ...appInfo.data, iconUrl: appsIconSvg }
+    if (appInfo.data.iconPath) {
+      try {
+        const iconInfo = await axios.get(`${appUrl}/${appInfo.data.iconPath}`)
```

Comment:
why this? can't you just put iconPath as src for `img`?

Reply:
it comes from another website

Reply:
👍